### DISCUSSION TITLE
When reordering the keys of an object with $entangle, Livewire doesn't actually overwrite the old object

### DISCUSSION THREAD

**Discussion by binaryfire** [livewire/livewire#7493] [7. Bugs (archived)]
### Livewire version

v3.2.6

### Which PHP version are you using?

PHP 8.2

### Repository URL

https://wirebox.app/b/y8232

### Steps To Reproduce

Issue reproduced in Wirebox: https://wirebox.app/b/y8232

Problem: When reordering the keys of an object with $entangle, or setting the entire object, Livewire doesn't actually overwrite the old object, so the keys remain in the same order.

This is causing the following issue in Filament: https://github.com/filamentphp/filament/issues/9030

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by binaryfire** [comment]
Can anyone help with writing a failing test for this?

**Reply by joshhanley** [reply]
@binaryfire can I suggest submitting the failing test yourself? The component you have working in wirebox is what you will need for the failing test. Just have a look at some of the other browser tests and you should be able to work out how to write the assertions.

Also we have just updated our contribution guide which should help get testing up and running, it's not live on the docs yet, but you can look at the details here https://github.com/livewire/livewire/blob/main/docs/contribution-guide.md

Thanks!

**Comment by binaryfire** [comment]
I can't get Dusk working on my system because of the headless Chrome config I use for other testing tools (it seems to conflict with Dusk). So if anyone can help with a failing test it would be appreciated. Would be great to get this bug fixed because it's breaking the Filament Key-Value field's sorting functionality. 

EDIT: I've added a failing test: https://github.com/livewire/livewire/pull/7671. But I couldn't test it locally so a maintainer might need to double-check / tweak it.
